src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -497,9 +497,10 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
 
         defaultDispatcher.dispatch<ViewRoomPayload>({
             action: Action.ViewRoom,
-            room_id: roomId,
             metricsTrigger: "WebUnifiedSearch",
             metricsViaKeyboard: viaKeyboard,
+            room_id: room.roomAlias ? undefined : room.roomId,
```

Comment:
You shouldn't be putting an alias `#` thing inside a field which only wants a `!` thing

The room_id field is never expected to not be an actual room id...

Reply:
I am?

Reply:
Oh I can't read, you should be sending the room_id unconditionally though, it saves an API call to look it up for trying to peek

Reply:
:+1: I concede I have always read it as exclusionary. I'm just going to be explicit with more words (and a rationale of why providing the room ID is useful).